scripts/c-to-cpp-includes.pl

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/perl
+
+# sweep over the squid repository, looking for patterns of
+# c standard library header files that have a c++ equivalent,
+# replacing these includes with the sequivalent standard c++ header.
+
+
+#header files to consider (without the .h extension)
+@headers=qw(
+assert
+ctype
+errno
+inttypes
+limits
+math
+signal
+stdarg
+stddef
+stdint
+stdio
+stdlib
+string
+time
+unistd
```

Comment:
I found a few more entries to add to the list: fenv, float, locale, setjmp, uchar, wchar, and wctype. Please note that it does not matter whether some particular Squid code snapshot actually uses them (all but one of these newly found ones are unused in recent master/v7 code): This script should "consider" _all_ possible C library header wrappers (that were not deprecated in C++20, as detailed below), so that we do not worry about script application timing.

AFAICT, those deprecated C library header wrappers correspond to the following tokens: stdbool, complex, tgmath, iso646, and stdalign. The script should _not_ replace the corresponding C headers with deprecated C++ headers IMO.

Disclaimer: I did not check whether some of these headers need special treatment that would preclude their automatic handling. I hope there are none, but if there are, then I would list them in a nearby script comment for completeness/clarity sake.

Reply:
Thanks for the additional digging. I agree with you, adding the extra
files to consider and mentioning the ones we won't in a comment